File: Data Structures & Algorithms/max-water-container/submission-0.py
class Solution:
    def maxArea(self, heights: List[int]) -> int:
        left, right = 0, len(heights) - 1
        max_l = 0

        while left < right:
            diff = right - left
            current = min(heights[right], heights[left]) * diff
            max_l = max(max_l, current)
            if heights[left] < heights[right]:
                left +=1
            else:
                right -= 1
        return max_l

File: Data Structures & Algorithms/binary-tree-diameter/submission-0.js
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     constructor(val = 0, left = null, right = null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    /**
     * @param {TreeNode} root
     * @return {number}
     */
    diameterOfBinaryTree(root) {
        let d = 0;

        function getHeight(node) {
            if (node == null) return 0;

            let lh = getHeight(node.left)
            let rh = getHeight(node.right)

            d = Math.max(d, lh + rh)

            return 1 + Math.max(lh, rh)
        }
        getHeight(root);

        return d;
    }
}
